### DISCUSSION TITLE
Controller Pak problems on N64

### DISCUSSION THREAD

**Discussion by xglacius** [darthcloud/BlueRetro#119] [Q&A]
Hi, I've configured v1.2 to use it on n64 with the NSO n64 controller. First I've use it with Rumble pak conguration which worked great. Now I want to use the Controller pak configuration. I've set it this way and tested it with Perfect Dark but it doesn't work properly.

First time the game detected a corrupted Controller pak which could be repaired and it worked OK. I've tried to save a game and it worked. I then powered cycle the console and the game didn't detected the controller pak. I have configured again with the web tool but the game sometimes detects it and sometimes not. 

Is there anything I can do? Thanks!

**Comment by darthcloud** [comment]
Did you format it first via web config? 

Plug it on a PC only via usb (no n64) and get the log:
https://github.com/darthcloud/BlueRetro/wiki/Getting-BlueRetro-debug-logs-via-Serial-port-Windows-10

While getting the log connect via web and format the pak.

Copy the whole logs in a text file and attach here (drag & drop in text box)

**Reply by darthcloud** [reply]
Also once formating done , wait 5sec the press reset (EN) button while log running.

Thanks

**Reply by xglacius** [reply]
Thanks for the quick reply. I've tried formatting and even reflashing and I've tried with a different game (Tony hawk pro skater) and sometimes it detects the controller pak and sometimes not.

I've attached the log you asked,
[blueretro.txt](https://github.com/darthcloud/BlueRetro/files/7735482/blueretro.txt)

**Comment by darthcloud** [comment]
Log look normal. Try this:

While your ESP32 only plug to PC via USB.

Unzip this file and write the .mpk file via the web pak manager.
[ctrl_pak1.zip](https://github.com/darthcloud/BlueRetro/files/7736266/ctrl_pak1.zip)

Then read back the pak and rename file to before_reboot.mpk

Reboot the adapter with EN button, connect to pak manager again and read the pak again, rename file after_reboot.pak

Zip the 2 files and attach to discussion.

This will help me figure if the data get corrupt, and if so  right away or only after reboot.

Thanks,
Jacques

**Reply by xglacius** [reply]
There you go!
[paks.zip](https://github.com/darthcloud/BlueRetro/files/7737338/paks.zip)

**Reply by darthcloud** [reply]
Ok their is no corruption in any of the files.

Does it work on N64 as it is now?

If not that would be problem at the N64 communication level. Can you share a picture of your setup where I can easily see the connections?

You got only one controller connection?

**Reply by xglacius** [reply]
I thought that it worked as it was at this point but not everytime. I tested it with perfect dark and it read it sometimes. Same thing with THPS, first time it said it was corrupted, second time worked fine. Then I've tested it with Snowboard kids and it directly says that no controller pak is inserted.

I've just one controller connection (see the pic attached). It's weird because rumble pack (when I had the other config) or buttons work perfect.

![IMG_20211218_100557](https://user-images.githubusercontent.com/96293461/146635940-f015082f-4ccd-4c98-968b-68cf77c57a15.jpg)

**Reply by darthcloud** [reply]
Wiring is fine, not sure why it doesn't for you. Maybe try another board? (get an official one here: https://www.digikey.ca/en/products/detail/espressif-systems/ESP32-DEVKITC-32E/12091810)

As a test, In the advanced config you could select one of the other banks. Note this setting affect only N64 side, so you need to format the pak via pak manager by selecting the same pak there.

**Reply by xglacius** [reply]
Thanks a lot for the ideas. I've tried with pak2 and same thing, sometimes it gets the pak and sometimes not. Maybe I'll buy another board to do more tests. I'll post them here!
